src/main/kotlin/io/fritz2/binding/seq.kt

```diff
@@ -1,54 +1,46 @@
-import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.flow.*
 import kotlinx.coroutines.flow.map
 
 data class Patch<out T>(val from: Int, val that: List<T>, val replaced: Int)
 
-class Seq<T>(initValue: List<T>,
-             private val channel: ConflatedBroadcastChannel<Patch<T>> = ConflatedBroadcastChannel<Patch<T>>(Patch(0, initValue,0)),
-             private val flow: Flow<Patch<T>> = channel.asFlow().distinctUntilChanged()) : Flow<Patch<T>> by flow {
-
-    private var last: List<T> = initValue
-    
-    fun value(): List<T> = last
-
-    //TODO: maybe better performance with iterable? zip?
-    @ExperimentalCoroutinesApi
-    suspend fun set(n: List<T>): Unit {
-//        console.log("### Seq: compare $n to $last")
-        val size2Compare = if (last.size < n.size) {
-//            console.log("### Seq: append " + (n.size - last.size) + " from " + (last.size))
-            channel.send(Patch(last.size, n.takeLast(n.size - last.size), 0))
-            last.size
-        }
-        else {
-            if (n.size < last.size) {
-//                console.log("### Seq: delete " + (last.size - n.size) + " from " + (n.size -1))
-                channel.send(Patch(n.size, listOf(), last.size - n.size ))
-            }
-            n.size
-        }
-
-//        console.log("### Seq: size to compare: $size2Compare")
-
-        //FIXME: better performance without range?
-        for (i in 0 until size2Compare) {
-//            console.log("### Seq: inner comparing: $i -> ${last[i]} to ${n[i]}")
-            if (last[i] != n[i]) channel.send(Patch(i, listOf(n[i]), 1))
+@ExperimentalCoroutinesApi
+private suspend inline fun <T> compare(oldAndNew: Pair<List<T>, List<T>>): Flow<Patch<T>> = channelFlow {
+    val (oldValue, newValue) = oldAndNew
+    val size2Compare = if (oldValue.size < newValue.size) {
+        //console.log("### Seq: append " + (newValue.size - oldValue.size) + " from " + (oldValue.size))
+        channel.send(Patch(oldValue.size, newValue.takeLast(newValue.size - oldValue.size), 0))
+        oldValue.size
+    } else {
+        if (newValue.size < oldValue.size) {
+            //console.log("### Seq: delete " + (oldValue.size - newValue.size) + " from " + (newValue.size -1))
+            channel.send(Patch(newValue.size, emptyList(), oldValue.size - newValue.size))
         }
+        newValue.size
+    }
+    //console.log("### Seq: size to compare: $size2Compare")
 
-        last = n
+    //FIXME: better performance without range?
+    for (i in 0 until size2Compare) {
+        //TODO: batch changed items in a row to one patch
+        //console.log("### Seq: inner comparing: $i -> ${last[i]} to ${n[i]}")
+        if (oldValue[i] != newValue[i]) channel.send(Patch(i, listOf(newValue[i]), 1))
     }
 }
 
+private suspend inline fun <T> accumulate(accumulator: Pair<List<T>, List<T>>, newValue: List<T>): Pair<List<T>, List<T>> = Pair(accumulator.second, newValue)
```

Comment:
Maybe add a comment here!?

Reply:
why? ok, have commented, but in an actual branch...